homeassistant/components/logbook/websocket_api.py

```diff
@@ -0,0 +1,304 @@
+            # system is overloaded with an event storm
+            await asyncio.sleep(EVENT_COALESCE_TIME)
+            while True:
+                try:
+                    events.append(stream_queue.get_nowait())
+                except asyncio.QueueEmpty:
+                    break
+
+            if logbook_events := event_processor.humanify(
+                async_event_to_row(e) for e in events
+            ):
+                connection.send_message(
+                    JSON_DUMP(
+                        messages.event_message(
+                            msg_id,
+                            logbook_events,
+                        )
+                    )
+                )
+        stream_queue.task_done()
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "logbook/event_stream",
+        vol.Required("start_time"): str,
+        vol.Optional("entity_ids"): [str],
+        vol.Optional("device_ids"): [str],
+    }
+)
+@websocket_api.async_response
+async def ws_event_stream(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """Handle logbook get events websocket command."""
+    start_time_str = msg["start_time"]
+    utc_now = dt_util.utcnow()
+
+    if start_time := dt_util.parse_datetime(start_time_str):
+        start_time = dt_util.as_utc(start_time)
+
+    if not start_time or start_time > utc_now:
+        connection.send_error(msg["id"], "invalid_start_time", "Invalid start_time")
+        return
+
+    device_ids = msg.get("device_ids")
+    entity_ids = msg.get("entity_ids")
+    if entity_ids:
+        entity_ids = async_filter_entities(hass, entity_ids)
+    event_types = async_determine_event_types(hass, entity_ids, device_ids)
+
+    event_processor = EventProcessor(
+        hass,
+        event_types,
+        entity_ids,
+        device_ids,
+        None,
+        timestamp=True,
+        include_entity_name=False,
+    )
+
+    stream_queue: asyncio.Queue[Event] = asyncio.Queue(MAX_PENDING_LOGBOOK_EVENTS)
+    subscriptions: list[CALLBACK_TYPE] = []
+    setup_complete_future: asyncio.Future[dt] = asyncio.Future()
+    task = asyncio.create_task(
+        _async_events_consumer(
+            setup_complete_future,
+            connection,
+            msg["id"],
+            stream_queue,
+            event_processor,
+        )
+    )
+
+    def _unsub() -> None:
+        """Unsubscribe from all events."""
+        for subscription in subscriptions:
+            subscription()
+        subscriptions.clear()
+        if task:
+            task.cancel()
+
+    @callback
+    def _queue_or_cancel(event: Event) -> None:
+        """Queue an event to be processed or cancel."""
+        try:
+            stream_queue.put_nowait(event)
+        except asyncio.QueueFull:
+            _LOGGER.debug(
+                "Client exceeded max pending messages of %s (likely disconnected without unsubscribe)",
```

Comment:
It's not possible to disconnect without unsubscribing because all unsubscriptions are automatically called on disconnect.

Reply:
I added test for this even but forgot to adjust the comment

done in a3019b581952bcd25adbe46bd66f769858db1af0